File: src/main/java/com/dg/watcher/base/Log.java
package com.dg.watcher.base;

import java.util.logging.Level;
import static java.util.logging.Level.INFO;
import static java.util.logging.Level.SEVERE;
import static java.util.logging.Level.WARNING;
import static java.util.logging.Logger.getLogger;


public class Log {
    public static void err(String tag, String msg) {
        log(SEVERE, tag, msg);
    }

    public static void warn(String tag, String msg) {
        log(WARNING, tag, msg);
    }

    public static void info(String tag, String msg) {
        log(INFO, tag, msg);
    }

    private static void log(Level level, String tag, String msg) {
        getLogger(tag).log(level, msg);
    }
}

File: src/main/java/com/dg/watcher/base/Log.kt
package com.dg.watcher.base

import java.util.logging.Level
import java.util.logging.Level.INFO
import java.util.logging.Level.SEVERE
import java.util.logging.Level.WARNING
import java.util.logging.Logger.getLogger


fun err(tag: String, msg: String) = log(SEVERE, tag, msg)

fun warn(tag: String, msg: String) = log(WARNING, tag, msg)

fun info(tag: String, msg: String) = log(INFO, tag, msg)

private fun log(level: Level, tag: String, msg: String) = getLogger(tag).log(level, msg)